requirements-dev.txt

```diff
@@ -1,6 +1,6 @@
 autopep8 # for code formatting
-pip # listed here so we can upgrade it as same time as everything else
 setuptools # for installing from source
+markupsafe==2.0.1 # needed due to sphinx having a sub-dependency on jinja2 but jinja2 does not install markupsafe
```

Comment:
I was getting the issue on `linux-compiler-compat (gcc-5)` and similar `ggc` and `clang` images: https://github.com/awslabs/aws-crt-python/actions/runs/4876356794/jobs/8699732427#step:2:349

It heavily depends on the version of Python you have installed on whether it will properly get MarkupSafe like expected or if it will have the issue in the CI link above. Telling it to manually install fixes it though.
Source:
* https://stackoverflow.com/questions/70316777/no-module-named-markupsafe-error-in-jetson-nano

Reply:
The comment about "jinja2 does not install markupsafe" isn't accurate. In the failed check you linked, it tried to install markupsafe and the installation failed for some reason. It wasn't a matter of dependencies being declared wrong.

I'm trying to reproduce this by running the `aws-crt-ubuntu-18-x64` docker image locally, but the installation always seems to work? I've tried installing it by itself, tried installing before and after upgrading pip, tried installing via `requirements-dev.txt` (with and without `pip` in the list)

Maybe it was just some side-effect of the state of this PR in the middle of the 32 commits?

I tried removing this line and pushed. Seems to work now 🤷‍♀️